app/controllers/initiatives_controller.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+class InitiativesController < ApplicationController
+  before_action :set_initiative, only: %i[show destroy update]
+
+  def index
+    @initiatives = Initiative.all
+  end
+
+  # POST
+  def create
+    @initiative = Initiative.new(initiative_params)
+    return if @initiative.save
+
+    render json: @initiative.errors, status: :unprocessable_entity
+  end
+
+  # PATCH/PUT
+  def update
+    return if @initiative.update(initiative_params)
+
+    render json: @initiative.errors, status: :unprocessable_entity
+  end
+
+  # DELETE
+  def destroy
+    @initiative.destroy
+  end
```

Comment:
This is missing the failure to destroy case

Reply:
You're right, addressed!